src/include/gurt/telemetry_consumer.h

```diff
@@ -49,12 +49,21 @@ int d_tm_list(struct d_tm_context *ctx, struct d_tm_nodeList_t **head,
 int d_tm_list_subdirs(struct d_tm_context *ctx, struct d_tm_nodeList_t **head,
 		      struct d_tm_node_t *node, uint64_t *node_count,
 		      int max_depth);
+
+typedef void (*d_tm_iter_cb_t)(struct d_tm_context *ctx, struct d_tm_node_t *node,
+			       int level, char *path, int format, int opt_fields,
+			       void *cb_arg);
+
 void d_tm_iterate(struct d_tm_context *ctx, struct d_tm_node_t *node,
 		  int level, int filter, char *path, int format,
-		  int opt_fields, uint32_t ops, FILE *stream);
+		  int opt_fields, d_tm_iter_cb_t iter_cb, void *cb_arg);
 void d_tm_print_node(struct d_tm_context *ctx, struct d_tm_node_t *node,
 		     int level, char *name, int format, int opt_fields,
 		     FILE *stream);
+
+void d_tm_reset_node(struct d_tm_context *ctx, struct d_tm_node_t *node, int level,
+		     char *path, int format, int opt_fields, FILE *stream);
```

Comment:
This shouldn't be available to the telemetry consumer, as this alters the data. If it needs to be accessible publicly, it belongs in telemetry_producer.h. It may not need to be public at all if you use ephemeral directories to handle the job IDs and pids.

Reply:
sure

Reply:
Hmm, I thought this maybe a consumer, since the user will use it to reset the value from daos_metrics utils? no?